types/mempool/priority_nonce.go

```diff
@@ -226,6 +216,16 @@ func (mp *PriorityNonceMempool[C]) InsertWithGasWanted(ctx context.Context, tx s
 	nonce := sig.Sequence
 	key := txMeta[C]{nonce: nonce, priority: priority, sender: sender}
 
+	mp.mtx.Lock()
+	defer mp.mtx.Unlock()
+	if mp.cfg.MaxTx > 0 && mp.priorityIndex.Len() >= mp.cfg.MaxTx {
```

Comment:
I think, the custom signer extractor is not as heavy as it was at the beginning, secondly, it's useful to do the `MaxTx` check as early as possible to reduce the cost when the traffic is very high like in a ddos attack scenario.